lib/services/local/users.go

```diff
@@ -193,6 +193,41 @@ func (s *IdentityService) UpdateUser(ctx context.Context, user types.User) error
 	return nil
 }
 
+// UpdateUserFunc reads an existing user, runs `fn` against it and writes the
+// result to storage. Return `false` from `fn` to avoid storage changes.
+// Roughly equivalent to [GetUser] followed by [CompareAndSwapUser].
+// Returns the storage user.
+func (s *IdentityService) UpdateUserFunc(ctx context.Context, user string, withSecrets bool, fn func(types.User) (changed bool, err error)) (types.User, error) {
```

Comment:
We've called it "update and swap" in PluginsService and in the generic local service, FYI.

Your name doesn't make it clear that it's an atomic modification, but "update and swap" doesn't make it clear that it's a closure doing the change, so I'm not sure which is better.

Reply:
Seems alright to me, renamed to `UpdateAndSwapUser`.